sandbox/cgroup/monitor.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"path"
+
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/sandbox/cgroup/inotify"
+	"github.com/snapcore/snapd/strutil"
+)
+
+// MonitorFullDelete allows to monitor a group of files/folders
+// and, when all of them have been deleted, emits the specified name through the channel.
+func MonitorFullDelete(name string, folders []string, channel chan string) error {
```

Comment:
A few things about this:

1. Is this meant to be a public API of the package? If not, call it `monitorFullDelete`. If you need access for testing, see the `export_test.go` file to see how we expose things.
2. In Go, it is usual to close channels from the sending side, since sending to a closed channel will panic, and closing a channel can useful to signal the receiving side. Who would close the channel in this situation, and when would it be safe to do so?
3. Are there situations where we'd want to interrupt the monitor before it completes?

Reply:
- Added mock function to avoid exporting monitorFullDelete.
- Added close(channel) to the tests. The channel must be closed by the caller when it receives the notification.
- No, as far as I can think...